File: simplerecyclerviewtouchlistener/src/main/java/com/wordpress/beendora/simplerecyclerviewtouchlistener/MyGestureListener.java
package com.wordpress.beendora.simplerecyclerviewtouchlistener;

import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

class MyGestureListener extends GestureDetector.SimpleOnGestureListener {

    private RecyclerView mRecyclerView;
    private SimpleRecyclerViewOnItemTouchListener.OnItemClickListener mOnItemClickListener;
    private SimpleRecyclerViewOnItemTouchListener.OnItemDoubleClickListener mOnItemDoubleClickListener;
    private SimpleRecyclerViewOnItemTouchListener.OnItemLongPressListener mOnItemLongPressListener;

    public MyGestureListener(RecyclerView mRecyclerView,
                             SimpleRecyclerViewOnItemTouchListener.OnItemClickListener mOnItemClickListener,
                             SimpleRecyclerViewOnItemTouchListener.OnItemDoubleClickListener mOnItemDoubleClickListener,
                             SimpleRecyclerViewOnItemTouchListener.OnItemLongPressListener mOnItemLongPressListener) {
        this.mRecyclerView = mRecyclerView;
        this.mOnItemClickListener = mOnItemClickListener;
        this.mOnItemDoubleClickListener = mOnItemDoubleClickListener;
        this.mOnItemLongPressListener = mOnItemLongPressListener;
    }

    @Override
    public boolean onSingleTapUp(MotionEvent e) {
        if (!(mRecyclerView == null)) {
            View view = mRecyclerView.findChildViewUnder(e.getX(), e.getY());
            if (!(view == null) && !(mOnItemClickListener == null))
                mOnItemClickListener.onClick(view, mRecyclerView.getChildAdapterPosition(view));
        }
        return true;
    }

    @Override
    public void onLongPress(MotionEvent e) {
        if (!(mRecyclerView == null)) {
            View view = mRecyclerView.findChildViewUnder(e.getX(), e.getY());
            if (!(view == null) && !(mOnItemClickListener == null))
                mOnItemLongPressListener.onLongPress(view, mRecyclerView.getChildAdapterPosition(view));
        }
    }

    @Override
    public boolean onDoubleTap(MotionEvent e) {
        if (!(mRecyclerView == null)) {
            View view = mRecyclerView.findChildViewUnder(e.getX(), e.getY());
            if (!(view == null) && !(mOnItemClickListener == null))
                mOnItemDoubleClickListener.onDoubleClick(view, mRecyclerView.getChildAdapterPosition(view));
        }
        return true;
    }

}

File: simplerecyclerviewtouchlistener/src/main/java/io/github/horaciocome1/simplerecyclerviewtouchlistener/MyGestureListener.kt
/*
 *    Copyright 2018 Horácio Flávio Comé Júnior
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific la
 */

package io.github.horaciocome1.simplerecyclerviewtouchlistener

import android.view.GestureDetector
import android.view.MotionEvent
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class MyGestureListener(val onClick: (View, Int) -> Unit,
                        val onDoubleClick: (View, Int) -> Unit,
                        val onLongPress: (View, Int) -> Unit,
                        private val recyclerView: RecyclerView) : GestureDetector.SimpleOnGestureListener() {

    override fun onSingleTapUp(e: MotionEvent?): Boolean {
        if (e != null) {
            val view = recyclerView.findChildViewUnder(e.x, e.y)
            if (view != null) onClick(view, recyclerView.getChildAdapterPosition(view))
        }
        return true
    }

    override fun onDoubleTap(e: MotionEvent?): Boolean {
        if (e != null) {
            val view = recyclerView.findChildViewUnder(e.x, e.y)
            if (view != null) onDoubleClick(view, recyclerView.getChildAdapterPosition(view))
        }
        return true
    }

    override fun onLongPress(e: MotionEvent?) {
        if (e != null) {
            val view = recyclerView.findChildViewUnder(e.x, e.y)
            if (view != null) onLongPress(view, recyclerView.getChildAdapterPosition(view))
        }
    }
}